### PR TITLE
fix: make desktop updates reliable

### PR DESCRIPTION
**Pull request by Bhavik-ag**
## Audit findings

- macOS in-app updates require a stable Developer ID signature. DBDesk has no Apple Developer account, so an unsigned or ad-hoc build cannot be made reliably self-updating with electron-updater.
- The Homebrew tap was stuck at v0.1.9: its scheduled job called GitHub CLI without `GH_TOKEN`, then assumed lowercase asset names that do not exist.
- Each platform published directly into the same release. The first completed job could expose an incomplete release, and the final release creation request returned HTTP 400 without failing the workflow.
- Renderer update listeners maintained separate transient state, so Settings, the user menu, and notifications could disagree or miss events.
- The old sidebar update card shown in the screenshot has already been removed on main; update actions now stay in the user menu, Settings, and toast.

## Changes

- Keep Windows and supported Linux packages on the centralized in-app updater state flow.
- Detect standard Homebrew-managed DBDesk installations. Settings refreshes Homebrew metadata and reports whether DBDesk is outdated, while leaving the actual upgrade to `brew upgrade`.
- Send unsigned macOS DMG installations to the Downloads page instead of offering a broken in-app updater.
- Build an ARM64 DMG with an ad-hoc signature, no notarization, and no Apple secrets; verify the bundle signature, architecture, and disk image in the release job.
- Build platform artifacts independently, keep the GitHub release as a draft, verify Windows/Linux update metadata and payload hashes, upload atomically, then publish.
- Add tag/version validation, package smoke tests, updater logging, duplicate-operation guards, and relaunch-on-install.
- Add a dedicated website update guide and update the repository installation guidance for macOS, Windows, AppImage, DEB, and Linux Homebrew.
- Fix the Homebrew tap in zexahq/homebrew-dbdesk#1 (merged).

## macOS transition

New installs should use:

```bash
brew install --cask zexahq/dbdesk/dbdesk
```

Updates use:

```bash
brew update
brew upgrade --cask zexahq/dbdesk/dbdesk
```

When an Apple Developer account is added later, switch the builder identity to Developer ID, restore hardened runtime/notarization plus the ZIP update artifact, add Apple signing secrets, and re-enable macOS in-app updates.

## Architecture

macOS is explicitly ARM64. A universal build is intentionally deferred because the bundled CLI currently includes an ARM64 native better-sqlite3 binary; labeling that bundle universal would leave the CLI broken on Intel.

## Verification

- `pnpm --dir apps/desktop run test` (58 tests)
- `pnpm run typecheck`
- `pnpm run lint`
- `pnpm run build`
- Real ARM64 macOS app and DMG build with explicit ad-hoc signing
- `codesign --verify --deep --strict` and architecture validation
- `hdiutil verify` on generated DMGs
- Formatting and `git diff --check`
- Homebrew cask/formula style and Ruby syntax checks in zexahq/homebrew-dbdesk#1

### CODE REVIEW

File: apps/web/src/app/(home)/components/download-button.tsx [hunk 1]
```diff
@@ -1,95 +1,95 @@
...
-  size: number;
+  name: string
+  browser_download_url: string
+  size: number
 }
 
 interface DownloadButtonProps {
-  assets?: Asset[];
+  assets?: Asset[]
 }
```
[the remark this answers is not in the document]
Comment (Bhavik-ag) [reply apps/web/src/app/(home)/components/download-button.tsx]: Valid. Replaced both non-extensible interfaces with type aliases in commit 8dcaa2f. Formatting, web lint, and the full monorepo typecheck pass.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -110,52 +133,96 @@ jobs:
...
+          CSC_KEY_PASSWORD: ${{ secrets.MAC_CSC_KEY_PASSWORD }}
+          APPLE_ID: ${{ secrets.APPLE_ID }}
+          APPLE_APP_SPECIFIC_PASSWORD: ${{ secrets.APPLE_APP_SPECIFIC_PASSWORD }}
+          APPLE_TEAM_ID: ${{ secrets.APPLE_TEAM_ID }}
         working-directory: apps/desktop
         run: |
           pnpm --dir ../../packages/cli run build
           pnpm exec electron-vite build
-          pnpm exec electron-builder --mac --publish always
+          pnpm exec electron-builder --mac --arm64 --publish never -c.mac.forceCodeSigning=true
```
[the remark this answers is not in the document]
Comment (Bhavik-ag) [reply .github/workflows/release.yml]: Resolved in the current branch: every macOS download label now says Apple silicon, and both the download page and repository documentation explicitly state that Intel Macs are unsupported.

File: apps/web/src/app/(home)/download/components/os-detect.tsx [hunk 3]
```diff
@@ -1,99 +1,99 @@
...
-  size: number;
+  name: string
+  browser_download_url: string
+  size: number
 }
 
 interface OSDetectProps {
-  assets?: Asset[];
+  assets?: Asset[]
 }
```
[the remark this answers is not in the document]
Comment (Bhavik-ag) [reply apps/web/src/app/(home)/download/components/os-detect.tsx]: Valid. Replaced both plain object interfaces with type aliases in commit 1deb2b4. Formatting, web lint, and the full monorepo typecheck pass.
